### PR TITLE
S1 T3: Implement logic for add user credentials

### PR DESCRIPTION
**Pull request by nancyprabhakar**
This PR is related to Story 1 Task 3: Implement logic for add user credentials.

I have created an endpoint for users to create an account. 
There are 2 types of users: Students and Service Staff. Each of these users have a different account creation experience (different input credentials are requested from the Student and the Service Staff).

This code first identifies if the user is a Student or a Service Staff. Secondly, it requests the user to input their credentials. For a student, these are:  
            firstName
            lastName
            email
            accessToken
            phone
            schoolID
For a Service Staff, these are: 
workerID
            firstName
            lastName
            email
            accessToken
            phone
            specialization

Third, it adds these users into the system.
This is based on the Student and Worker objects.

### CODE REVIEW

**Comment by catai9** [comment]
Hi Nancy,
I reviewed your code and I listed some recommendations below.

1. I would recommend starting with S1 T1 first to create the stub to ensure that both front end and back end are aware of the parameters required for this method and the desired output. 
2. Did you try to compile this code using npm start? It looks like Java syntax rather than JavaScript.
3. The user input is handled by the front end rather than the backend. The endpoint would take in those parameters when updating the respective tables.

File: server/models/handlers/signup.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+           
+           //Create arrays to store the credentials provided by the user 
+        //Array for users who are students
+        ArrayList<userType> users = new ArrayList<userType>(); 
+        //Array for users who are serviceStaff
+        ArrayList<userType> serviceStaff = new ArrayList<userType>(); 
+            
+            //Create a scanner to injest the credentials which the user inputs
+
+            Scanner input = new Scanner(System.in);
```
Comment (catai9) [inline server/models/handlers/signup.js]: Not sure why scanner is being used here. The user input is handled by the front end through the sign in form. The endpoint would only need to take in those values to perform queries.

File: server/models/handlers/signup.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    System.out.print("firstName : "); 
+    email = input.next();
+
+    System.out.print("lastName : "); 
+    email = input.next();
+
+    System.out.print("email : "); 
+    email = input.next();
+
+    System.out.print("accessToken
```
Comment (catai9) [inline server/models/handlers/signup.js]: The user would not know their access token and this value would need to be generated randomly using an npm library such as randomstring (https://www.npmjs.com/package/randomstring)

File: server/models/handlers/signup.js (hunk 1 above)
Comment (nancyprabhakar) [reply server/models/handlers/signup.js]: Hi Amy, could you point me to some resources I can reference on inputting parameters from the front end sign up form?

File: server/models/handlers/signup.js (hunk 2 above)
Comment (nancyprabhakar) [reply server/models/handlers/signup.js]: I see, I'll change this so an authentication token is generated as an output of this logic.

Comment (nancyprabhakar) [reply server/models/handlers/signup.js]: Made the requested changes!

File: server/models/handlers/signup.js (hunk 1 above)
Comment (nancyprabhakar) [reply server/models/handlers/signup.js]: Would the user need to select a password and input that through the sign up form?

Comment (Kyle199910) [reply server/models/handlers/signup.js]: im assumming so, since password is needed for a user account, both for signing up and logging in to the system

**Comment by Kyle199910** [comment]
Hi Nancy,

Like with what Amy said, code should focus on the logic of getting user inputted data from the front end, and updating the backend tables with the new data. Also, another thing is that the code should be written in a j.s(javascript) file in express, not java.

**Comment by catai9** [comment]
Closing as the code needs to be rewritten in the proper language (JavaScript NOT Java).

**Pull request closed without merging**